File: Bindings~/java/src/main/java/com/alttester/Commands/InputActions/AltBeginTouch.java
/*
    Copyright(C) 2026 Altom Consulting
*/

package com.alttester.Commands.InputActions;

import com.alttester.Commands.AltBaseCommand;
import com.alttester.IMessageHandler;

public class AltBeginTouch extends AltBaseCommand {
  private AltBeginTouchParams params;

  public AltBeginTouch(IMessageHandler messageHandler, AltBeginTouchParams params) {
    super(messageHandler);
    this.params = params;
  }

  public int Execute() {
    SendCommand(params);
    String data = recvall(params, String.class);
    return Integer.parseInt(data);
  }
}

File: Assets/AltTester/Runtime/AltDriver/Commands/InputActions/AltBeginTouch.cs
/*
    Copyright(C) 2026 Altom Consulting
*/

namespace AltTester.AltTesterSDK.Driver.Commands
{
    public class AltBeginTouch : AltBaseCommand
    {
        AltBeginTouchParams cmdParams;

        public AltBeginTouch(IDriverCommunication commHandler, AltVector2 coordinates) : base(commHandler)
        {
            this.cmdParams = new AltBeginTouchParams(coordinates);
        }
        public int Execute()
        {
            CommHandler.Send(cmdParams);
            return CommHandler.Recvall<int>(cmdParams);  //finger id
            //TODO: add handling for "Finished"
        }
    }
}
